CGPs/0021.md

```diff
@@ -0,0 +1,36 @@
+# CGP [0021]: Update randomnessBlockRetentionWindow to Extend Attestation Expiration Duration
+
+- Date: 2021-01-29
+- Author(s): @anna-carroll
+- Status: DRAFT
+- Governance Proposal ID #: [if submitted]
+- Date Executed: [if executed]
+- Discussions To: https://forum.celo.org/t/discussions-for-cgp-21-update-randomnessblockretentionwindow-to-extend-attestation-expiration-duration/752
+
+## Overview
+
+This proposal will effectively extend the expiration of a requested attestation from 1 HOUR -> 1 DAY, improving upon [CGP 0015](https://github.com/celo-org/celo-proposals/blob/master/CGPs/0015.md) which previously attempted to do so.
+
+The initial [CGP 0015](https://github.com/celo-org/celo-proposals/blob/master/CGPs/0015.md) attempted to extend the expiration time of attestations from 1 hour to 24 hours by changing attestationExpiryBlocks in the Attestation contract. However, the Random contract only retains randomness for a period of blocks specified by randomnessBlockRetentionWindow. Though the Attestation contract was updated to be able to query randomness between 1 and 24 hours, the Random contract remains unable to serve those requests. Therefore, the expiration time for attestations remains functionally 1 hour.
```

Comment:
Thank you for submitting your first CGP :) 

Here's the changes I recommend for this paragraph:
* attestationExpiryBlocks -> `attestationExpiryBlocks`
* randomnessBlockRetentionWindow -> `randomnessBlockRetentionWindow`
* Attestation -> `Attestations.sol`
* Random -> `Random.sol`

Reply:
https://github.com/celo-org/celo-proposals/pull/155/commits/b73cbc76624e95c338f0ca2fadf09b26517f08f7